### PR TITLE
python3Packages.gcsfs: init at 2021.04.0

### PR DESCRIPTION
**Pull request by nbren12**
###### Motivation for this change

Add gcsfs, a useful tool for working with Google Cloud Storage. gcsfs implements the interfaces described in fsspec, which is in nixpkgs. I needed to bump the versions fsspec to work with the latest release of gcsfs. These packages are developed in unison, so usually need very similar versions.

I sometimes contribute to this project and am in contact with its primary author.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by legendofmiracles** [comment]
Make another commit for adding the maintainer `maintainers: add nbren12`

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6977,6 +6977,12 @@
     githubId = 818502;
     name = "Nathan Yong";
   };
+  nbren12 = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:6980]: Please cherry-pick the commit into the PR where you are adding yourself as a maintainer of a package.

File: pkgs/development/python-modules/fsspec/default.nix [hunk 2]
```diff
@@ -1,28 +1,23 @@
-{ lib
-, buildPythonPackage
-, fetchFromGitHub
-, pythonOlder
-, pytestCheckHook
-, numpy
-, stdenv
-}:
+{ lib, buildPythonPackage, fetchFromGitHub, pythonOlder, pytestCheckHook, numpy
+, stdenv, aiohttp, pytest-vcr, requests }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fsspec/default.nix:2]: Please revert this change to keep the diff smaller.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please include the version number in the init commit.

File: pkgs/development/python-modules/fsspec/default.nix [hunk 3]
```diff
@@ -33,6 +28,9 @@ buildPythonPackage rec {
     # resolved in https://github.com/intake/filesystem_spec/issues/432 and
     # can be enabled again from version 0.8.4
     "test_pathobject"
```
Comment (dotlambda) [inline pkgs/development/python-modules/fsspec/default.nix:30]: Can this be removed?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120173` at 023c823f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.clifford</li>
    <li>python38Packages.dask-jobqueue</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.rl-coach</li>
    <li>python39Packages.dask-jobqueue</li>
    <li>python39Packages.dask-xgboost</li>
    <li>python39Packages.rl-coach</li>
  </ul>
</details>
<details>
  <summary>16 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.caffe</li>
    <li>python38Packages.dask-glm</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.imgaug</li>
    <li>python38Packages.intake</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.caffe</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>54 packages built successfully:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.batchgenerators</li>
    <li>python38Packages.dask</li>
    <li>python38Packages.dask-gateway</li>
    <li>python38Packages.dask-image</li>
    <li>python38Packages.dask-mpi</li>
    <li>python38Packages.datasets</li>
    <li>python38Packages.distributed</li>
    <li>python38Packages.fsspec</li>
    <li>python38Packages.gcsfs</li>
    <li>python38Packages.glymur</li>
    <li>python38Packages.image-match</li>
    <li>python38Packages.imagecorruptions</li>
    <li>python38Packages.pims</li>
    <li>python38Packages.pyfftw</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.radio_beam</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.s3fs</li>
    <li>python38Packages.scikitimage</li>
    <li>python38Packages.slicedimage</li>
    <li>python38Packages.sparse</li>
    <li>python38Packages.streamz</li>
    <li>python38Packages.stumpy</li>
    <li>python38Packages.stytra</li>
    <li>python38Packages.tensorly</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.batchgenerators</li>
    <li>python39Packages.dask</li>
    <li>python39Packages.dask-gateway</li>
    <li>python39Packages.dask-image</li>
    <li>python39Packages.dask-mpi</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.distributed</li>
    <li>python39Packages.fsspec</li>
    <li>python39Packages.gcsfs</li>
    <li>python39Packages.glymur</li>
    <li>python39Packages.image-match</li>
    <li>python39Packages.imagecorruptions</li>
    <li>python39Packages.pims</li>
    <li>python39Packages.pyfftw</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.radio_beam</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.s3fs</li>
    <li>python39Packages.scikitimage</li>
    <li>python39Packages.slicedimage</li>
    <li>python39Packages.streamz</li>
  </ul>
</details>

**Comment by legendofmiracles** [comment]
Trying to build this has now frozen my computer twice, because of what I cannot tell.

But the build log before freezing has 2 errors.

**Comment by nbren12** [comment]
Result of `nixpkgs-review pr 120173` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.clifford</li>
    <li>python38Packages.dask-jobqueue</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python39Packages.dask-jobqueue</li>
    <li>python39Packages.dask-xgboost</li>
  </ul>
</details>
<details>
  <summary>11 packages failed to build:</summary>
  <ul>
    <li>python38Packages.dask-glm</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.intake</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.streamz</li>
    <li>python38Packages.stumpy</li>
    <li>python39Packages.dask-glm</li>
    <li>python39Packages.intake</li>
    <li>python39Packages.spectral-cube</li>
    <li>python39Packages.streamz</li>
  </ul>
</details>
<details>
  <summary>53 packages built:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.batchgenerators</li>
    <li>python38Packages.dask</li>
    <li>python38Packages.dask-gateway</li>
    <li>python38Packages.dask-image</li>
    <li>python38Packages.dask-mpi</li>
    <li>python38Packages.datasets</li>
    <li>python38Packages.distributed</li>
    <li>python38Packages.fsspec</li>
    <li>python38Packages.gcsfs</li>
    <li>python38Packages.glymur</li>
    <li>python38Packages.image-match</li>
    <li>python38Packages.imagecorruptions</li>
    <li>python38Packages.pims</li>
    <li>python38Packages.pyfftw</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.radio_beam</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.s3fs</li>
    <li>python38Packages.scikitimage</li>
    <li>python38Packages.slicedimage</li>
    <li>python38Packages.sparse</li>
    <li>python38Packages.stytra</li>
    <li>python38Packages.sunpy</li>
    <li>python38Packages.tensorly</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.batchgenerators</li>
    <li>python39Packages.dask</li>
    <li>python39Packages.dask-gateway</li>
    <li>python39Packages.dask-image</li>
    <li>python39Packages.dask-mpi</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.distributed</li>
    <li>python39Packages.fsspec</li>
    <li>python39Packages.gcsfs</li>
    <li>python39Packages.glymur</li>
    <li>python39Packages.image-match</li>
    <li>python39Packages.imagecorruptions</li>
    <li>python39Packages.pims</li>
    <li>python39Packages.pyfftw</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.radio_beam</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.s3fs</li>
    <li>python39Packages.scikitimage</li>
    <li>python39Packages.slicedimage</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>

File: pkgs/development/python-modules/fsspec/default.nix (hunk 2 above)
Comment (nbren12) [reply pkgs/development/python-modules/fsspec/default.nix:2]: done.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (nbren12) [reply maintainers/maintainer-list.nix:6980]: I split this change into a separate commit: e0b51cc7ae4e59a389781ab373c9acd1d78ae017.


> Please cherry-pick the commit into the PR

Confused about the wording here. Do you mean "a new PR" or "the PR"? For now, I will assume that "The PR" refers to this one.

File: pkgs/development/python-modules/fsspec/default.nix (hunk 3 above)
Comment (nbren12) [reply pkgs/development/python-modules/fsspec/default.nix]: I removed it. The issue has indeed been resolved on the fsspec side.

**Comment by nbren12** [comment]
@legendofmiracles @SuperSandro2000 

I force-pushed a new batch of commits, splitting the maintainer addition into a new commit, and addressing the other comments.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix:6980]: I think I picked the wrong default answer.

The correct one is:

Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: nixos/modules/installer/tools/tools.nix [hunk 4]
```diff
@@ -163,7 +163,7 @@ in
         # List packages installed in system profile. To search, run:
         # \$ nix search wget
         # environment.systemPackages = with pkgs; [
-        #   nano vim # don't forget to add an editor to edit configuration.nix!
+        #   vim # Do not forget to add an editor to edit configuration.nix! The Nano editor is also installed by default.
         #   wget
```
Comment (SuperSandro2000) [inline nixos/modules/installer/tools/tools.nix:167]: I think you included this commit by accident. Please drop it.

Comment (nbren12) [reply nixos/modules/installer/tools/tools.nix]: Weird. I re-based onto master and this diff went away.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please add `pythonImportsCheck = [ "gcsfs" ]`.

File: pkgs/development/python-modules/gcsfs/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+buildPythonPackage rec {
+  pname = "gcsfs";
+  version = "2021.04.0";
+
+  # github sources needed for test data
+  src = fetchFromGitHub {
+    owner = "dask";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-qwLi4+rNSAneQbbq2ljf58jDluVo6b/2SUwUNpFb1/Y=";
```
Comment (dotlambda) [inline pkgs/development/python-modules/gcsfs/default.nix:14]: Hash is wrong.

Comment (nbren12) [reply pkgs/development/python-modules/gcsfs/default.nix]: Weird. I wonder if upstream moved the tag to another commit. I've updated it.

Comment (dotlambda) [reply pkgs/development/python-modules/gcsfs/default.nix]: You should ask them. If I did, that would be reason enough for me not to allow this package into nixpkgs.

Comment (nbren12) [reply pkgs/development/python-modules/gcsfs/default.nix]: IMO this package is professionally maintained. The hashing error was my mistake. Github shows the tag was last updated before I wrote my original commit.

**Review by dotlambda — APPROVED** [review]
